src/App.js

```diff
@@ -1,19 +1,62 @@
-import React from 'react';
-import './App.css';
+import React, { useEffect, useState } from 'react';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import './App.css';
 
 const App = () => {
+
+  const [chatEntryData, setChatEntryData] = useState([
+    {
+    id: 0,
+    sender: '',
+    body: '',
+    timeStamp: '',
+    liked: false
+    }
+  ]);
+
+  useEffect(() => {
+    setChatEntryData(chatMessages);
+  }, []);
+  
+  const [likesCount, setLikesCount] = useState(0);
+
+  const updateChatEntryData = updatedEntry => {
+    const updatedEntries = chatEntryData.map(entry => {
+      if (entry.id === updatedEntry.id) {
+        if (updatedEntry.liked === true) {
+          setLikesCount((likesCount) => likesCount + 1);
```

Comment:
Since the total number of likes can be found without introducing state, this line can be removed.